prov/efa/src/rxr/rxr_msg.c

```diff
@@ -445,7 +445,7 @@ int rxr_msg_match_unexp(struct dlist_entry *item, const void *arg)
 
 	rx_entry = container_of(item, struct rxr_rx_entry, entry);
 
-	return rxr_match_addr(match_info->addr, rx_entry->addr);
+	return rxr_match_addr(rx_entry->addr, match_info->addr);
```

Comment:
I can see that this change is correct, but why do not we need it before?

Reply:
Before the patch that changed this, we had the FI_DIRECTED_RECV check in both in `rxr_generic_recv` and when searching for a rx_entry given an unexpected message.